pkg/k8s/apis/cilium.io/utils/utils.go

```diff
@@ -63,34 +63,10 @@ func GetPolicyLabels(ns, name string) labels.LabelArray {
 	)
 }
 
-// ParseToCiliumRule returns an api.Rule with all the labels parsed into cilium
-// labels.
-func ParseToCiliumRule(namespace, name string, r *api.Rule) *api.Rule {
-	retRule := &api.Rule{}
-	if r.EndpointSelector.LabelSelector != nil {
-		retRule.EndpointSelector = api.NewESFromK8sLabelSelector("", r.EndpointSelector.LabelSelector)
-		// The PodSelector should only reflect to the same namespace
-		// the policy is being stored, thus we add the namespace to
-		// the MatchLabels map.
-		if retRule.EndpointSelector.LabelSelector.MatchLabels == nil {
-			retRule.EndpointSelector.LabelSelector.MatchLabels = map[string]string{}
-		}
-
-		userNamespace, ok := retRule.EndpointSelector.LabelSelector.MatchLabels[podPrefixLbl]
-		if ok && userNamespace != namespace {
-			log.WithFields(logrus.Fields{
-				logfields.K8sNamespace:              namespace,
-				logfields.CiliumNetworkPolicyName:   name,
-				logfields.K8sNamespace + ".illegal": userNamespace,
-			}).Warn("CiliumNetworkPolicy contains illegal namespace match in EndpointSelector." +
-				" EndpointSelector always applies in namespace of the policy resource, removing illegal namespace match'.")
-		}
-		retRule.EndpointSelector.LabelSelector.MatchLabels[podPrefixLbl] = namespace
-	}
-
-	if r.Ingress != nil {
-		retRule.Ingress = make([]api.IngressRule, len(r.Ingress))
-		for i, ing := range r.Ingress {
+func parseToCiliumIngressRule(namespace string, inRule, retRule *api.Rule) {
```

Comment:
Why not return the rule instead of passing it as a parameter to the function?

Reply:
This is because the same rule is used throughout `ParseToCiliumRule`. I could make the function have the rule as its receiver?